src/main/java/org/jvnet/hudson/test/junit/jupiter/JenkinsRuleExtension.java

```diff
@@ -0,0 +1,36 @@
+package org.jvnet.hudson.test.junit.jupiter;
+
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.ParameterContext;
+import org.junit.runner.Description;
+import org.jvnet.hudson.test.JenkinsRecipe;
+import org.jvnet.hudson.test.JenkinsRule;
+
+import javax.annotation.Nonnull;
+import java.lang.reflect.Method;
+import java.util.Optional;
+
+/**
+ * Provides JUnit 5 compatibility for {@link JenkinsRule}.
+ */
+public class JenkinsRuleExtension extends JenkinsRule {
+	private final ParameterContext context;
+
+	JenkinsRuleExtension(@Nonnull ParameterContext context, @Nonnull ExtensionContext extensionContext) {
```

Comment:
Change to support the earlier suggestion for spotbugs NonNull instead of JSR-305 Nonnull.
```suggestion
	JenkinsRuleExtension(@NonNull ParameterContext context, @NonNull ExtensionContext extensionContext) {
```